reflex/state.py

```diff
@@ -218,24 +262,39 @@ def __init__(
         *args,
         parent_state: BaseState | None = None,
         init_substates: bool = True,
+        _reflex_internal_init: bool = False,
         **kwargs,
     ):
         """Initialize the state.
 
+        DO NOT INSTANTIATE STATE CLASSES DIRECTLY! Use StateManager.get_state() instead.
+
         Args:
             *args: The args to pass to the Pydantic init method.
             parent_state: The parent state.
             init_substates: Whether to initialize the substates in this instance.
+            _reflex_internal_init: A flag to indicate that the state is being initialized by the framework.
             **kwargs: The kwargs to pass to the Pydantic init method.
 
+        Raises:
+            RuntimeError: If the state is instantiated directly by end user.
         """
+        if not _reflex_internal_init and not _is_testing_env():
```

Comment:
I wonder if instead of checking `_is_testing_env` we should check the inverse, `is_reflex_run`. So that when external users work with it, they don't have to set this flag (similar to the hack I had to do in flexdown)

Reply:
lets talk about this on monday, i'm not really grokking the suggestion. how does `is_reflex_run` get determined?